contrib/llvm-project-cmake/CMakeLists.txt

```diff
@@ -21,6 +21,9 @@ set (LLVM_INCLUDE_DIRS
     "${ClickHouse_BINARY_DIR}/contrib/llvm-project/llvm/include"
 )
 set (LLVM_LIBRARY_DIRS "${ClickHouse_BINARY_DIR}/contrib/llvm-project/llvm")
+# NOTE: You should not remove this line since otherwise it will use default 20,
+# and llvm cannot be compiled with bundled libcxx and 20 standard.
+set (CMAKE_CXX_STANDARD 14)
```

Comment:
Why not 17 as it was before?

Reply:
Because official llvm uses 14.

Reply:
Do I look at the wrong place? https://github.com/llvm/llvm-project/blob/main/llvm/CMakeLists.txt#L63

Reply:
Kind off - https://github.com/ClickHouse/llvm-project/blob/bd0ba013354bbfb8db03eeb9b553d629b58f65ac/llvm/CMakeLists.txt#L65

Although previously 17 was used, but 14 here because w/o cmake cache 14 was used, IOW all builds now uses 14 standard.

Reply:
LLVM 15 (which ClickHouse uses) compiles C++14, unreleased LLVM HEAD/16 compiles with C++17.

I actually hit the failure in "MasmParser.cpp.o" a few times by myself when I upgraded LLVM. But I did not care further because I thought something with my system is wrong ^^ So thanks for fixing this.